### DISCUSSION TITLE
Update user email address in auth.users table?

### DISCUSSION THREAD

**Discussion by GodGardensGuns** [supabase/supabase#573] [Questions] [answered]
Is there any way for users to update the email addresses they use to log in? I would like to use email sign-up/log-in within my app and use the same provided by the user for a number of other functions. If I could synchronize user email changes across all features that require it (including Supabase), that would help a lot.

**Comment by carlosstenzel** [comment, marked as answer]
I tested an SQL code and it worked

```
update auth.users set email = 'newemail@codehamper.com' where id = 'ID_USER'
```

**Reply by GodGardensGuns** [reply]
Thank you! I'm new to both SQL and Supabase and didn't know if there was an exposed function in the Supabase SDK that did the same.

**Reply by GodGardensGuns** [reply]
I created a stored procedure (aka function) in the database that I hoped would allow the user's email address to be updated from the supabase JS client. When I ran it, I was met with an error message: 'permission denied for schema auth'

Here's what my 'CREATE FUNCTION' query looks like:
```
CREATE FUNCTION updateauthemail (new_email text, auth_id text)
RETURNS text AS $updated_data$
	DECLARE
		updated_data text;
	BEGIN
		UPDATE auth.users set email = new_email where id = auth_id;
		RETURN updated_data;
END;
$updated_data$ LANGUAGE plpgsql;
```

Do I need to add something to the function when creating it, or pass a certain param through the JS client on my Node.js server? Here's the function I'm calling through the JS client:
```
const { data: rpcData, error: rpcErr } = await supabase.rpc(
      "updateauthemail",
      {
        auth_id: authId,
        new_email: updatedData.email,
      }
    );
```

**Reply by inian** [reply]
Can you try adding `security definer` to the create function definition?
```
...
END;
$updated_data$ LANGUAGE plpgsql security definer;
```

**Reply by GodGardensGuns** [reply]
Thanks @inian. Wasn't sure if this was a separate problem so I opened an issue where @kiwicopple pointed me towards past issues that answered my question. See: https://github.com/supabase/supabase-js/issues/127#issuecomment-777269488
